autoPyTorch/pipeline/components/setup/network/base_network.py

```diff
@@ -106,24 +111,35 @@ def predict(self, loader: torch.utils.data.DataLoader) -> torch.Tensor:
         """
         Performs batched prediction given a loader object
         """
-        assert self.network is not None
-        self.network.eval()
-
+        if len(self.network_snapshots) == 0:
+            assert self.network is not None
+            return self._predict(network=self.network, loader=loader).cpu().numpy()
```

Comment:
I think also here. we dont need to push it to the cpu.

Reply:
True, I removed it.